components/certificate-mgt/org.wso2.carbon.identity.certificate.management/src/main/java/org/wso2/carbon/identity/certificate/management/dao/impl/CacheBackedApplicationCertificateMgtDAO.java

```diff
@@ -0,0 +1,113 @@
+import org.wso2.carbon.identity.certificate.management.dao.ApplicationCertificateManagementDAO;
+import org.wso2.carbon.identity.certificate.management.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.model.Certificate;
+
+/**
+ * This class represents the caching layer of Application Certificate Management.
+ * This supports using auto-incremented IDs for certificate management operations in application-mgt.
+ *
+ * @deprecated It is recommended to use {@link CacheBackedCertificateMgtDAO}, which supports operations with UUID.
+ */
+@Deprecated
+public class CacheBackedApplicationCertificateMgtDAO implements ApplicationCertificateManagementDAO {
+
+    private static final Log LOG = LogFactory.getLog(CacheBackedApplicationCertificateMgtDAO.class);
+    private final CertificateCacheById certificateCacheById;
+    private final ApplicationCertificateManagementDAO applicationCertificateMgtDAO;
+
+    public CacheBackedApplicationCertificateMgtDAO(ApplicationCertificateManagementDAO applicationCertificateMgtDAO) {
+
+        this.applicationCertificateMgtDAO = applicationCertificateMgtDAO;
+        certificateCacheById = CertificateCacheById.getInstance();
+    }
+
+    @Override
+    @Deprecated
+    public int addCertificate(String certificateId, Certificate certificate, int tenantId)
+            throws CertificateMgtException {
+
+        return applicationCertificateMgtDAO.addCertificate(certificateId, certificate, tenantId);
+    }
+
+    @Override
+    @Deprecated
+    public Certificate getCertificate(int certificateId, int tenantId) throws CertificateMgtException {
+
+        CertificateIdCacheKey cacheKey = new CertificateIdCacheKey(String.valueOf(certificateId));
+        CertificateCacheEntry entry = certificateCacheById.getValueFromCache(cacheKey, tenantId);
+
+        if (entry != null) {
+            LOG.debug("Cache entry found for Certificate integer id: " + certificateId);
+            return entry.getCertificate();
+        }
+
+        LOG.debug("Cache entry not found for Certificate integer id: " + certificateId +
+                ". Fetching entry from DB.");
+
+        Certificate certificate = applicationCertificateMgtDAO.getCertificate(certificateId, tenantId);
+
+        if (certificate != null) {
+            LOG.debug("Entry fetched from DB for Certificate integer id: " + certificateId + ". Updating cache.");
+            certificateCacheById.addToCache(cacheKey, new CertificateCacheEntry(certificate), tenantId);
+        } else {
+            LOG.debug("Entry for Certificate integer id: " + certificateId + " not found in cache or DB.");
+        }
+        return certificate;
+    }
+
+    @Override
+    @Deprecated
+    public Certificate getCertificateByName(String certificateName, int tenantId) throws CertificateMgtException {
+
+        Certificate certificate = applicationCertificateMgtDAO.getCertificateByName(certificateName, tenantId);
```

Comment:
reading from cache is not done because this method is deprecated ?

Reply:
If so, does the current application management component caches this or not.
if caching layer is missing there as well this is fine

Reply:
We are caching against the certificate id. In this case we don't have the id reference. Instead we have the certificate name. Hence we cannot utilize the cache in here.